sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -534,3 +534,44 @@ abstract class GpuLastBase(child: Expression)
   override lazy val deterministic: Boolean = false
   override def toString: String = s"gpulast($child)${if (ignoreNulls) " ignore nulls"}"
 }
+
+/**
+ * Collects and returns a list of non-unique elements.
+ *
+ * The two 'offset' parameters are not used by GPU version, but are here for the compatibility
+ * with the CPU version and automated checks.
+ */
+case class GpuCollectList(child: Expression,
+                          mutableAggBufferOffset: Int = 0,
+                          inputAggBufferOffset: Int = 0)
+  extends GpuDeclarativeAggregate with GpuAggregateWindowFunction {
+
+  def this(child: Expression) = this(child, 0, 0)
+
+  override lazy val deterministic: Boolean = false
```

Comment:
nit: Could we add a comment similar to what is in the Spark code for this.

```
  // Both `CollectList` and `CollectSet` are non-deterministic since their results depend on the
  // actual order of input rows.
```

Reply:
Thanks for review. Added it.